### PR TITLE
All meal addition/deletion updates persist across refreshes - including snacks.

### PR DESCRIPTION
**Pull request by izaxveta**
#### Pivotal URL:
https://www.pivotaltracker.com/story/show/154856788
https://www.pivotaltracker.com/story/show/154615186
https://www.pivotaltracker.com/story/show/154615188

#### What does this PR do?
This PR updates all relevant calorie totals (total calories/remaining calories) when a food is either added or removed from a meal. This PR also takes care of any changes to meal tables persisting between refreshes by making a fetch request to update the API.
 
#### Where should the reviewer start?
Start at the addNewFoodsToMeal function in the index.js file.

#### Any additional context you want to provide?
🐞🐞🐞 This PR also fixes the bug related to snacks. Code has been modified to update 'snacks' from plural to singular in order to be consistent where exactly 'snack' is required for proper functionality.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -46,42 +46,65 @@ $(document).ready(() => {
     if (event.target.nodeName == 'BUTTON') {
       var foodId = event.target.id.split('-')[2]
       var mealName = $(event.target).attr('data').split('-')[0]
-      foodsDiary.deleteFoodFromMeal(mealName, foodId)
+      foodsDiary.updateMealWithFood(mealName, foodId, 'DELETE')
```
Comment (marshallhouston) [inline lib/index.js:49]: Nice passing in the method and making the fetch call dynamic!

File: lib/index.js [hunk 2]
```diff
@@ -96,7 +119,8 @@ const addNewFoodsToMeal = () => {
     var food = setFoodData(checkedFoods[index])
     var meal = $(event.target).attr('data')
     foodsDiary.renderFoodToMealTable(meal, food)
-    updateMealCalories(food, meal)
+    updateMealCalories(food, meal, 'increaseMeal')
+    foodsDiary.updateMealWithFood(meal, food.id, 'POST')
```
Comment (marshallhouston) [inline lib/index.js:123]: 🎉 for making these functions more dynamic.

File: lib/index.js [hunk 3]
```diff
@@ -46,42 +46,65 @@ $(document).ready(() => {
...
+  }
+  var totalCalories = parseInt($(totalCaloriesNode).text())
+  var foodCalories = parseInt(food.calories)
+  var updatedTotalCalories = calculateTotals(operator, totalCalories, foodCalories)
   updateNodeValue(totalCaloriesNode, updatedTotalCalories)
 }
 
+const calculateTotals = (operator, digit1, digit2) => {
+  var formulate = {
+    '+': function (digit1, digit2) { return digit1 + digit2 },
```
Comment (marshallhouston) [inline lib/index.js]: This function declaration is in ES5 syntax.

Will you change it to ES6? I pulled down the branch and found that this is a way to use ES6 syntax and keep functionality.
```
'+' : (digit1, digit2) => { return digit1 + digit2 },
'-' : (digit1, digit2) => { return digit1 - digit2 }
```

Here's a link to the documentation about it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions

**Review by marshallhouston — CHANGES_REQUESTED** [review]
Awesome work on making these functions handle either increase or decrease and making sure changes persist across refreshes!

I've requested changes on 2 lines that use ES5 syntax. Once those are changed, I'll approve.

File: lib/index.js (hunk 3 above)
Comment (izaxveta) [reply lib/index.js]: Good catch! Syntax has been updated to ES6.

**Comment by izaxveta** [comment]
Function has been updated to incorporate ES6 syntax.

File: lib/index.js [hunk 4]
```diff
@@ -99,8 +99,8 @@ const updateTotalCaloriesValue = (food, totalCaloriesNode, method) => {
 
 const calculateTotals = (operator, digit1, digit2) => {
   var formulate = {
-    '+': function (digit1, digit2) { return digit1 + digit2 },
-    '-': function (digit1, digit2) { return digit1 - digit2 }
+    '+' : (digit1, digit2) => { return digit1 + digit2 },
```
Comment (marshallhouston) [inline lib/index.js]: 👍

**Review by marshallhouston — APPROVED** [review]

**Pull request merged**